### DISCUSSION TITLE
Is it possible to pass postgres func like "now()" in args?

### DISCUSSION THREAD

**Discussion by a1tus** [jackc/pgx#1722] [Q&A] [answered]
Hello. Query example:

`UPDATE table SET field=$1, updated_at=$2 WHERE id=$3`

is it possible to pass `NOW()` via `$2`?

Of course, in this simplified example it can be hardcoded, but it's need when I want to use time from app OR from db conditionally.
Also, we use `pgxutil.UpdateRow` where sql query is not used at all.

Thanks.

**Comment by jackc** [comment, marked as answer]
Your first example is not possible. At the PostgreSQL protocol level a bound parameter is a value not a SQL expression.

With `pgxutil.UpdateRow` it is not possible _at the present_. I've wanted that feature myself, but have not yet determined a good solution. Hopefully in the future.

**Comment by a1tus** [comment]
We are trying to use only db time in such fields to get predictable times that don't depend on app servers.
`pgxutil.UpdateRow` is great, it simplified many functions but caused this issue.
Thanks for fast reply, we'll try to make some workaround.

**Comment by jackc** [comment]
I just added `pgxutil.SQLValue` type to `pgxutil`. It can be used with the `Insert*` and `Update*` functions.

e.g. 

```go
pgxutil.InsertRow(ctx, conn, "people", map[string]any{"name": "John", "updated_at": pgxutil.SQLValue("now()")})
```

I don't anticipate changing this feature, but continue to bear in mind that pgxutil is an experimental package and may change.

**Reply by a1tus** [reply]
Wow! That would be definately useful, many thanks!
